environment/aurora/main.tf

```diff
@@ -0,0 +1,88 @@
+resource "aws_rds_cluster" "cluster" {
+  count                           = var.aurora_serverless ? 0 : 1
+  apply_immediately               = var.apply_immediately
+  availability_zones              = var.availability_zones
+  backup_retention_period         = var.backup_retention_period
+  cluster_identifier              = "${var.cluster_identifier}-${terraform.workspace}"
+  database_name                   = var.database_name
+  db_subnet_group_name            = var.db_subnet_group_name
+  deletion_protection             = var.deletion_protection
+  engine                          = var.engine
+  engine_version                  = var.engine_version
+  enabled_cloudwatch_logs_exports = ["postgresql"]
+  final_snapshot_identifier       = "${var.database_name}-${terraform.workspace}-final-snapshot"
+  kms_key_id                      = var.kms_key_id
+  master_username                 = var.master_username
+  master_password                 = var.master_password
+  preferred_backup_window         = "05:15-05:45"
+  preferred_maintenance_window    = "mon:05:50-mon:06:20"
+  replication_source_identifier   = var.replication_source_identifier
+  skip_final_snapshot             = var.skip_final_snapshot
+  storage_encrypted               = var.storage_encrypted
+  vpc_security_group_ids          = var.vpc_security_group_ids
+  tags                            = var.tags
+
+  lifecycle {
+    ignore_changes  = [replication_source_identifier]
+    prevent_destroy = true
+  }
+}
+
+resource "aws_rds_cluster_instance" "cluster_instances" {
+  count                           = var.aurora_serverless ? 0 : var.instance_count
+  auto_minor_version_upgrade      = var.auto_minor_version_upgrade
```

Comment:
I can see we default this to false but does the terraform workflow update that var at all?

Reply:
Probably phrased that a bit weird...I guess what I really need to know is if the var value is static or dynamic (similar to tfvars etc).

Reply:
It's static, It will only change if `auto_minor_version_upgrade` is passed into the module call. I would leave it static while we migrate then it should be a simple change once we've migrated.